torch/_inductor/codegen/wrapper.py

```diff
@@ -2582,7 +2582,9 @@ def generate_kernel_call(
         ), f"cuda kernel parameters for {name} should already exist at this moment"
         mangled_name = params.get("mangled_name", None)
         assert mangled_name is not None, "missing mangled_name"
-        cubin_path = params.get("cubin_path", None)
+        cubin_path = params.get(
+            "cubin_path" if torch.version.hip is None else "hsaco_path", None
```

Comment:
Considering that you need exactly the same in codecache, I wonder if we can turn it into a helper function... (like `get_cubin_path_name`)

Reply:
Seems sensible looks like I'm thinking `cpp_wrapper_cubin_path` inline with the `cpp_wrapper_cache_dir` naming here: https://github.com/pytorch/pytorch/blob/main/torch/_inductor/codecache.py#L114

I'll update the PR with all suggested changes shortly

Reply:
Adopted this suggestion. @malfet does this look okay?
[get_cpp_wrapper_cubin_path_name](https://github.com/pytorch/pytorch/blob/6749d0208958c86a0057bca7ca1e7a73bb2a404d/torch/_inductor/codecache.py#L129C5-L129C36)

```
def get_cpp_wrapper_cubin_path_name():
    return "cubin_path" if torch.version.hip is None else "hsaco_path"
```

Reply:
@malfet In regards to `cuda_to_hip_mappings` change.

This feature will generate a C++ file which uses CUDA to launch triton kernels from cubin binaries. To support this on ROCm we are hipifying the actual codegen logic in wrapper.py to generate an equivalent HIP cpp wrapper code.

Here is a gist with the hipified file contents:
https://gist.github.com/jataylo/df6cf593e204e5196b64c7e47e056083